opencensus/stats/exporters/prometheus_exporter.py

```diff
@@ -345,4 +347,11 @@ def get_view_name(namespace, view):
     name = ""
     if namespace != "":
         name = namespace + "_"
-    return name + view.name
+    return sanitize(name + view.name)
+
+
+def sanitize(key):
+    """ sanitize the given metric name or label according to Prometheus rule.
+    Replace all characters other than [A-Za-z0-9_] with '_'.
+    """
+    return re.sub("[^a-zA-Z0-9_]", "_", key)
```

Comment:
Suggest to pre-compile the regular expression for performance given this will be applied for all columns.

For example: https://github.com/census-instrumentation/opencensus-python/blob/master/opencensus/trace/tracestate.py#L25